From linux-usb-devel-bounces@lists.sourceforge.net Tue Aug 21 00:22:31 2007
From: Paul Walmsley <paul@pwsan.com>
Date: Thu, 16 Aug 2007 16:21:35 -0700
Subject: usb quirks: Add Canon EOS 5D (PC Connection mode) to the autosuspend blacklist
To: linux-kernel@vger.kernel.org
Cc: gphoto-devel@lists.sourceforge.net, Raj Kumar <rkumar@archive.org>, Brewster Kahle <brewster@archive.org>, Siznax <steve@archive.org>
Message-ID: <20070816232239.533779000@pwsan.com>
Content-Disposition: inline; filename=canon-eos5d-no-autosuspend.patch


Recent versions of the Linux kernel auto-suspend attached USB devices.
After this happens to the Canon EOS 5D camera, the camera's interrupt endpoints
don't seem to wake back up correctly, causing further use with libgphoto2
to fail with a -114 "OS error in camera communication" error.

A similar fix is probably necessary for this camera in PTP mode, which
identifies as USB product id 0x3102, but we haven't tested this.

As part of our testing process, we tried the USB_QUIRK_RESET_RESUME
quirk also, it's not helpful in this case.

Signed-off-by: Raj Kumar <rkumar@archive.org>
Signed-off-by: Paul Walmsley <paul@pwsan.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/core/quirks.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -79,6 +79,9 @@ static const struct usb_device_id usb_qu
 	/* Alcor multi-card reader */
 	{ USB_DEVICE(0x058f, 0x6366), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 
+	/* Canon EOS 5D in PC Connection mode */
+	{ USB_DEVICE(0x04a9, 0x3101), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+
 	/* RIM Blackberry */
 	{ USB_DEVICE(0x0fca, 0x0001), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 	{ USB_DEVICE(0x0fca, 0x0004), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
